CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/submission/SubmissionManager.java

```diff
@@ -72,84 +69,80 @@ public static void mongoInsertSubmission(final DB dbs, final String uniqueId, fi
-        if (corsor == null) {
+        if (cursor == null) {
             LOG.info("Creating a new instance to this old itemid");
             queryObj.append(SELF_ID, new ObjectId(problemId));
             collection.insert(queryObj);
-            // we need to create a new corsor
+            // we need to create a new cursor
         } else {
             LOG.info("adding a new submission to this old itemid");
             // insert the submissionId, if it is an experiment then we need to
             // use the uniqueId to make it work.
-            collection.update(corsor, new BasicDBObject("$set", queryObj));
+            collection.update(cursor, new BasicDBObject("$set", queryObj));
         }
     }
 
     /**
      * Sends a request to the submission server to request an experiment as a user.
+     *
      * @param dbs The database that contains data about the experiment.
      * @param userId The user who has access to the experiment.
      * @param problemId The id of the problem associated with the sketch.
-     * @param sessionInfo The session information that is sent to the submission server.
-     * @param internalConnections A manager of connections to another database.
+     * @param submissionManager The connections of the submission server
      * @throws DatabaseAccessException Thrown is there is data missing in the database.
+     * @throws AuthenticationException Thrown if the user does not have the authentication
+     * @return {@link protobuf.srl.submission.Submission.SrlExperiment} that had the specific submission id.
      */
-    public static void mongoGetExperiment(final DB dbs, final String userId, final String problemId, final Request sessionInfo,
-            final MultiConnectionManager internalConnections) throws DatabaseAccessException {
-
-        final ItemRequest.Builder build = ItemRequest.newBuilder();
-        build.setQuery(ItemQuery.EXPERIMENT);
-        final DBRef myDbRef = new DBRef(dbs, EXPERIMENT_COLLECTION, new ObjectId(problemId));
-        final DBObject corsor = myDbRef.fetch();
-        if (corsor == null) {
+    public static Submission.SrlExperiment mongoGetExperiment(final DB dbs, final String userId, final String problemId,
+            final SubmissionManagerInterface submissionManager) throws DatabaseAccessException, AuthenticationException {
+
+        final Data.ItemResult.Builder send = Data.ItemResult.newBuilder();
+        send.setQuery(ItemQuery.EXPERIMENT);
+        final DBObject cursor = dbs.getCollection(DatabaseStringConstants.EXPERIMENT_COLLECTION).findOne(new ObjectId(problemId));
+        if (cursor == null) {
             throw new DatabaseAccessException("The student has not submitted anything for this problem");
         }
-        final String sketchId = "" + corsor.get(userId);
-        LOG.info("SketchId: ", sketchId);
-        if ("null".equals(sketchId)) {
+        if (!cursor.containsField(userId) || Strings.isNullOrEmpty((String) cursor.get(userId))) {
```

Comment:
add `cursor == null` to this if statement and delete the preceding if statement cause they do the same thing